Projects/Server/Text/TextEncoding.cs

```diff
@@ -114,33 +115,55 @@ public static int GetByteLengthForEncoding(this Encoding encoding) =>
             _          => 1
         };
 
-    [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    private static bool IsSafeChar(ushort c) => c is >= 0x20 and < 0xFFFE;
-
     public static string GetString(ReadOnlySpan<byte> span, Encoding encoding, bool safeString = false)
     {
-        string s = encoding.GetString(span);
-
         if (!safeString)
         {
-            return s;
+            return encoding.GetString(span);
         }
 
-        ReadOnlySpan<char> chars = s.AsSpan();
+        var charCount = encoding.GetMaxCharCount(span.Length);
```

Comment:
We use GetMaxCharCount instead of GetCharCount since the latter is considerably slower and sometimes allocates.